st2common/st2common/ssh/parallel_ssh.py

```diff
@@ -0,0 +1,212 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import traceback
+
+import eventlet
+
+from st2common import log as logging
+from st2common.ssh.paramiko_ssh import ParamikoSSHClient
+import st2common.util.jsonify as jsonify
+
+LOG = logging.getLogger(__name__)
+
+
+class ParallelSSHClient(object):
+    KEYS_TO_TRANSFORM = ['stdout', 'stderr']
+    CONNECT_ERROR = 'Cannot connect to host.'
+
+    def __init__(self, hosts, user=None, password=None, pkey=None, port=22, concurrency=10,
+                 raise_on_error=False, connect=True):
+        self._ssh_user = user
+        self._ssh_key = pkey
+        self._ssh_password = password
+        self._hosts = hosts
+        self._ssh_port = port
```

Comment:
Kinda answering my question from above myself - it looks like that the code now expects the same SSH server on all the hosts.

Imo, we should still support `<host>:<port>,<host>:<port>` notation.